news/11617.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Deprecated a historical ambiguity in how ``egg`` fragments in URL-style
+requirements are formatted and handled. ``egg`` fragments that do not look
+like PEP 508 names now produce a deprecation warning.
```

Comment:
N.B.: I tagged this `NEWS` entry as `bugfix` instead of `removal` since it's a deprecation of unintended and not an outright removal, but let me know if that's the wrong category.

Reply:
The section is deprecation and removals, so `removal` is OK. :)

Reply:
Fixed, thanks!